File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        s, e = 0, len(nums) - 1

        while s < e:
            mid = s + (e - s) // 2
            if nums[mid] < nums[e]:
                e = mid
            else:
                s = mid + 1

        return nums[s]

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> _set;
        for (auto num : nums) {
            if (_set.count(num) == 1)
                return true;
            else
                _set.insert(num);
        }
        return false;
    }
};